### PR TITLE
raw-dataset-principal-investigator

### PR DESCRIPTION
**Pull request by nitrosx**
Added principalInvestigator field for raw dataset and changed investigator field only for derived dataset

### CODE REVIEW

**Comment by jl-wynen** [comment]
This was an intentional change to remove the duplication. Why do we need to distinguish them?

**Comment by nitrosx** [comment]
SciCat uses principal investigator for raw dataset and investigator for derived dataset.
It would be confusing using a different naming in scitacean

**Comment by jl-wynen** [comment]
My idea was to hide that fact from users. It is confusing for people like us. But do you think it is also confusing for people who never see behind the frontend and Scitacean?

**Comment by nitrosx** [comment]
In SciCat FE, raw datasets have Principal Investigator field, while derived datasets have Investigator.
Technically it is correct the difference in name, I understand it might be confusing.
Unfortunately, if we mask this difference in this library, users will be confused when checking directly SciCat FE.

**Review by jl-wynen — CHANGES_REQUESTED** [review]
Ok.

Then please implement this change via `tools/mode-generation/spec/dataset.yml` and name the Scitacean field `principal_investigator`.
And fix the broken tests and probably docs.

**Comment by nitrosx** [comment]
i would also suggest a change in name: from meta to metadata... but we can talk about it first

**Comment by nitrosx** [comment]
@jl-wynen Would you mind checking the errors that I get and point me in the right direction, please? 
I'm not sure I understand how model works.

File: tools/model-generation/spec/dataset.yml [hunk 1]
```diff
@@ -43,18 +43,25 @@ fields:
...
     type: List[PID]
     used: [ derived ]
   - name: investigator
-    description: Email of the (principal) investigator. The string may contain a list of emails, which should then be separated by semicolons.
-    model_name: { derived: investigator, raw: principalInvestigator }
+    description: Name(s) of the investigator(s). The string may contain a list of names, which should then be separated by semicolons.
+    model_name: investigator
     required: true
     type: str
-    validation: emails
```
Comment (jl-wynen) [inline tools/model-generation/spec/dataset.yml:57]: Does this PR contain everything from #63?

File: tools/model-generation/spec/dataset.yml [hunk 2]
```diff
@@ -43,18 +43,25 @@ fields:
...
     description: Array of input dataset identifiers used in producing the derived dataset. Ideally these are the global identifier to existing datasets inside this or federated data catalogs.
     model_name: inputDatasets
     required: true
     type: List[PID]
     used: [ derived ]
   - name: investigator
-    description: Email of the (principal) investigator. The string may contain a list of emails, which should then be separated by semicolons.
-    model_name: { derived: investigator, raw: principalInvestigator }
+    description: Name(s) of the investigator(s). The string may contain a list of names, which should then be separated by semicolons.
+    model_name: investigator
```
Comment (jl-wynen) [inline tools/model-generation/spec/dataset.yml:55]: Can be omitted since the names are identical

File: src/scitacean/_dataset_fields.py [hunk 3]
```diff
@@ -579,6 +590,7 @@ def __init__(
             "instrument_group": _apply_default(instrument_group, None, None),
             "instrument_id": _apply_default(instrument_id, None, None),
             "investigator": _apply_default(investigator, None, None),
+            "principalInvestigator": _apply_default(principalInvestigator, None, None),
```
Comment (jl-wynen) [inline src/scitacean/_dataset_fields.py:593]: This looks wrong. Why is this still camelCase?

File: tests/dataset_fields_test.py [hunk 4]
```diff
@@ -309,7 +309,7 @@ def test_make_raw_model():
         creationTime=dateutil.parser.parse("2142-04-02T16:44:56"),
         owner="Ponder Stibbons;Mustrum Ridcully",
         ownerGroup="faculty",
-        principalInvestigator="p.stibbons@uu.am",
+        principal_investigator="my principal investigator",
```
Comment (jl-wynen) [inline tests/dataset_fields_test.py:312]: You got the names here and in line 302 the wrong way around

**Review by jl-wynen — CHANGES_REQUESTED** [review]

File: tools/model-generation/spec/dataset.yml (hunk 2 above)
Comment (nitrosx) [reply tools/model-generation/spec/dataset.yml:55]: What is the preferred/suggested method? With or without?

File: tools/model-generation/spec/dataset.yml (hunk 1 above)
Comment (nitrosx) [reply tools/model-generation/spec/dataset.yml:57]: No. If we merged #63, this should be resolved.

File: tests/dataset_fields_test.py (hunk 4 above)
Comment (nitrosx) [reply tests/dataset_fields_test.py:312]: Do you mind explaining? 
I'm confused which one is the model retrieved from SciCat and which one if the scitacean object.
Thanks

File: tools/model-generation/spec/dataset.yml (hunk 2 above)
Comment (jl-wynen) [reply tools/model-generation/spec/dataset.yml:55]: I would keep it brief and not list `model_name`. But it doesn't really matter

File: tests/dataset_fields_test.py (hunk 4 above)
Comment (jl-wynen) [reply tests/dataset_fields_test.py]: Look at where the classes are imported from. `RawDataset` comes from `model` and is what goes through the API. `Dataset` comes from `scitacean` directly (strictly speaking `scitacean.dataset`) and is the user facing class with Scitacean's naming scheme.

File: src/scitacean/_dataset_fields.py [hunk 5]
```diff
@@ -590,7 +590,7 @@ def __init__(
             "instrument_group": _apply_default(instrument_group, None, None),
             "instrument_id": _apply_default(instrument_id, None, None),
             "investigator": _apply_default(investigator, None, None),
-            "principalInvestigator": _apply_default(principalInvestigator, None, None),
```
Comment (jl-wynen) [inline src/scitacean/_dataset_fields.py]: Line 539 above is still wrong

**Pull request merged**